backend/types.py

```diff
@@ -133,15 +134,13 @@ class EmbedderConfig(ModelConfig):
     pass
 
 
-class ParserConfig(BaseModel):
+class ParserConfig(ModelConfig):
     """
     Parser configuration
     """
 
-    parser: str = Field(title="Parser Class Name")
-    kwargs: Optional[Dict[str, Any]] = Field(
-        title="Parser Class Arguments", default_factory=dict
-    )
+    type: ModelType = ModelType.parser
+    pass
```

Comment:
Uh, I would rather just create another model and enum called ModuleConfig and ModuleType with the same structure

ModelConfig can envolve in future with model specific things

Reply:
Kinda debating where does Embedder fit?
Embedder itself is a module but emedding models should be Model

Maybe then we reserve ModelConfig for model objects in the gateway class

Reply:
For now, I'll keep it as is in the PR, maybe we can decide in some other PR on what do do.